docs/customizations.md

```diff
@@ -0,0 +1,292 @@
+# Grow Up Story for Generated SDKs
+
+This quickstart will introduce how to group up your generated code with customizations.
+
+**Note: This quickstart does not focus on renaming or removing generated objects**. If you'd like to rename or remove generated objects, the changes should really be made in the
+OpenAPI definition of your service. If you absolutely can not modify the OpenAPI definition, you can use directives, specifically [these pre-defined directives](https://github.com/Azure/autorest/blob/main/docs/generate/built-in-directives.md) in your README configuration
+
+## Key Concept: \_patch.py
+
+The `_patch.py` files at each level of the modules will be the entry point to customize the generated code.
+
+For example, if you want to override a model, you will use the `_patch.py` file at the `models` level of your
+generated code to override.
+
+The main flow of the `_patch.py` file will be:
+
+1. Import the generated object you wish to override. Generated objects will have the suffix `Generated`
+2. Inherit from the generated object, and override its behavior with your desired code functionality
+3. Include the name of your customized object in the `__all__` of the `_patch.py` file
+
+To test that your behavior has been properly customized, please add tests for your customized code.
+If you find that your customizations for an object are not being called, please make sure that
+your customized object is included in the `__all__` of your `_patch.py` file.
+
+The `_patch.py` file will never be removed during regeneration, so no worries about your customizations being
+lost!
+
+## Examples
+
+- [Improve Documentation](#improve-documentation)
+- [Change Model Behavior](#change-model-behavior)
+- [Add a model](#add-a-model)
+- [Change Operation Behavior](#change-operation-behavior)
+- [Overload an Operation](#overload-an-operation)
+- [Change Client Behavior](#change-client-behavior)
+- [Add a Client Method](#add-a-client-method)
+
+### Improve Documentation
+
+Documentation for generated objects can be easily improved in the `_patch.py` by overriding generated code.
+
+In this example, we improve the docstrings for model `Foo` by adding more information.
+
+In this `_patch.py` file:
+
+```
+azure-sdk
+│   README.md
+│
+└───azure
+    └───sdk
+        └───models
+        │   _models.py # where the generated models are
+        |   _patch.py # where we customize the models code
+```
+
+```python
+from ._models import FooGenerated
```

Comment:
One thing writing this doc made me realize: we're really solidifying the file names of the generated code. I think it would be great (for version tolerant at least) to have the following names
1. `_models.py` for models (we won't really need 2.7 models any more)
2. `_operations.py` for operations (we already do this by default for version tolerant)
3. `_client.py` for client (currently it's the name of the service client)